acstools/acszpt.py

```diff
@@ -0,0 +1,379 @@
+
+    @property
+    def detector(self):
+        """The user supplied detector. (str)"""
+        return self._detector
+
+    @property
+    def filt(self):
+        """The user supplied filter, if one was given. (str or `None`)"""
+        return self._filt
+
+    @property
+    def zpt_table(self):
+        """The results returned by the ACS Zeropoint Calculator. (`astropy.table.QTable`)"""
+        return self._zpt_table
+
+    def _check_inputs(self):
+        """Check the inputs to ensure they are valid.
+
+        Returns
+        -------
+        status : bool
+            True if all inputs are valid, False if one is not.
+
+        """
+        valid_detector = True
+        valid_filter = True
+        valid_date = True
+        # Determine the submitted detector is valid
+        if self.detector not in self._valid_detectors:
+            msg = ('{} is not a valid detector option.\n'
+                   'Please choose one of the following:\n{}\n'
+                   '{}'.format(self.detector,
+                               '\n'.join(self._valid_detectors),
+                               self._msg_div))
+            LOG.error(msg)
+            valid_detector = False
+
+        # Determine if the submitted filter is valid
+        if (self.filt is not None and
+                self.filt not in self.valid_filters[self.detector]):
+            msg = ('{} is not a valid filter for {}\n'
+                   'Please choose one of the following:\n{}\n'
+                   '{}'.format(self.filt, self.detector,
+                               '\n'.join(self.valid_filters[self.detector]),
+                               self._msg_div))
+            LOG.error(msg)
+            valid_filter = False
+
+        # Determine if the submitted date is valid
+        date_check = self._check_date()
+        if date_check is not None:
+            LOG.error('{}\n{}'.format(date_check, self._msg_div))
+            valid_date = False
+
+        if not valid_detector or not valid_filter or not valid_date:
+            return False
+
+        return True
+
+    def _check_date(self, fmt='%Y-%m-%d'):
+        """Convenience method for determining if the input date is valid.
+
+        Parameters
+        ----------
+        fmt : str
+            The format of the date string. The default is ``%Y-%m-%d``, which
+            corresponds to ``YYYY-MM-DD``.
+
+        Returns
+        -------
+        status : str or `None`
+            If the date is valid, returns `None`. If the date is invalid,
+            returns a message explaining the issue.
+
+        """
+        result = None
+        try:
+            dt_obj = dt.datetime.strptime(self.date, fmt)
+        except ValueError:
+            result = '{} does not match YYYY-MM-DD format'.format(self.date)
+        else:
+            if dt_obj < self._acs_installation_date:
+                result = ('The observation date cannot occur '
+                          'before ACS was installed ({})'.
```

Comment:
I am not a fan of separating the `.` and the `format()` like this as it is easy to miss in debugging.